### ISSUE TITLE
Culler raises an exception for a hub deployed at a different base URL

### ISSUE THREAD

**Issue by betatim** [jupyterhub/jupyterhub#2004]
Freshly deployed vanilla hub using the zero2jupyterhub chart (v0.7-8b9cb79) on google cloud. Noen standard parts are that this uses an nginx-ingress to route traffic to the hub which is at example.com/foobar.

When the culler runs the following error appears. I suspect the "ends with f" is somehow related to the base URL being `/foobar` (I did pick exactly this name for the test deployment it isn't a placeholder).
```
[E 180621 12:33:35 ioloop:792] Exception in callback functools.partial(<function wrap.<locals>.null_wrapper at 0x7fd7bd74f950>, <Future finished exception=HTTP 599: Port number ended with 'f'>)
    Traceback (most recent call last):
      File "/usr/local/lib/python3.6/dist-packages/tornado/ioloop.py", line 759, in _run_callback
        ret = callback()
      File "/usr/local/lib/python3.6/dist-packages/tornado/stack_context.py", line 276, in null_wrapper
        return fn(*args, **kwargs)
      File "/usr/local/lib/python3.6/dist-packages/tornado/ioloop.py", line 780, in _discard_future_result
        future.result()
      File "/usr/local/lib/python3.6/dist-packages/tornado/gen.py", line 1107, in run
        yielded = self.gen.throw(*exc_info)
      File "/usr/local/bin/cull_idle_servers.py", line 111, in cull_idle
        resp = yield fetch(req)
      File "/usr/local/lib/python3.6/dist-packages/tornado/gen.py", line 1099, in run
        value = future.result()
      File "/usr/local/lib/python3.6/dist-packages/tornado/gen.py", line 1113, in run
        yielded = self.gen.send(value)
      File "/usr/local/bin/cull_idle_servers.py", line 105, in fetch
        return (yield client.fetch(req))
      File "/usr/local/lib/python3.6/dist-packages/tornado/gen.py", line 1099, in run
        value = future.result()
    tornado.curl_httpclient.CurlError: HTTP 599: Port number ended with 'f'
```

Will test/debug a bit more and add details.

**Comment by chicocvenancio** [comment]
@betatim are you certain the `base_url` is set as `/foobar` and not `foobar`? I would expect this error if the base_url is set without the starting slash.

**Comment by minrk** [comment]
Which helm chart complains about the absolute base URL? I just successfully launched the z2jh chart with `hub.baseUrl = 'thiswontwork'` without complaint.

This is exactly what would happen if base_url started with 'f' instead of '/', so if your config value is correct, then it's probably something in the chart. Also note that base_url must always *start and end* with '/', so `/foobar` is also invalid.

**Comment by betatim** [comment]
Using v0.7-8b9cb79 of the chart.

`hub.baseUrl= 'foobar'` leads to `Error: UPGRADE FAILED: Ingress.extensions "jupyterhub" is invalid: spec.rules[0].http.paths[0].path: Invalid value: "foobar": must be an absolute path` when I `helm upgrade ...`

With a starting and trailing `/` `helm upgrade ...` works.

Was scrolling through the logs just now and couldn't find my original error message from above. But the logs are full of the below (baseUrl = /staginghub in this case, and chart version v0.7-e2757f0):
```
[W 2018-06-22 11:46:45.186 JupyterHub log:158] 404 GET /staginghub/hub/staginghubhub/api/users (cull-idle@127.0.0.1) 11.02ms
[E 180622 11:46:45 ioloop:792] Exception in callback functools.partial(<function wrap.<locals>.null_wrapper at 0x7fba6eacd158>, <Future finished exception=HTTP 404: Not Found>)
    Traceback (most recent call last):
      File "/usr/local/lib/python3.6/dist-packages/tornado/ioloop.py", line 759, in _run_callback
        ret = callback()
      File "/usr/local/lib/python3.6/dist-packages/tornado/stack_context.py", line 276, in null_wrapper
        return fn(*args, **kwargs)
      File "/usr/local/lib/python3.6/dist-packages/tornado/ioloop.py", line 780, in _discard_future_result
        future.result()
      File "/usr/local/lib/python3.6/dist-packages/tornado/gen.py", line 1107, in run
        yielded = self.gen.throw(*exc_info)
      File "/usr/local/bin/cull_idle_servers.py", line 111, in cull_idle
        resp = yield fetch(req)
      File "/usr/local/lib/python3.6/dist-packages/tornado/gen.py", line 1099, in run
        value = future.result()
      File "/usr/local/lib/python3.6/dist-packages/tornado/gen.py", line 1113, in run
        yielded = self.gen.send(value)
      File "/usr/local/bin/cull_idle_servers.py", line 105, in fetch
        return (yield client.fetch(req))
      File "/usr/local/lib/python3.6/dist-packages/tornado/gen.py", line 1099, in run
        value = future.result()
    tornado.httpclient.HTTPError: HTTP 404: Not Found
```

After a few minutes of waiting after a relaunch after setting the baseUrl to `/foo/` the error does not appear in either versions of the chart. So I think the original issue here is user error, maybe we can make the chart validate the baseUrl as starting and ending with a /?

**Comment by minrk** [comment]
Yup, if the helm chart is validating prefixes, it might as well validate them all the way, checking for both starting and ending slashes, which are both required.

I think the proper fix is #2008 and remove the workaround from the helm chart. JupyterHub itself validates base url config and adds leading/trailing slashes if they are missing, to be sure that behavior is always consistent. The bug here is due to the helm chart taking a low-level config value itself and bypassing the Hub's own normalizing.

**Comment by manics** [comment]
I'm closing this as it's quite old and there's no concrete action point. If you're still having problems please post on the [Jupyter Community Forum](https://discourse.jupyter.org/) or open a new issue. Thanks!
